### PR TITLE
OTel Upgrade to v.1.21.0

### PR DESCRIPTION
**Pull request by brunobat**
We skipped 1.20.x because it was a bit unstable. No Big changes since 1.19.
Some aws libs needed for tests were moved to contrib. In the future we will rewrite those tests to get rid of the dependencies.

### CODE REVIEW

**Review by radcortez — APPROVED** [review]

**Review by michalvavrik — COMMENTED** [review]
Thanks for this!

File: bom/application/pom.xml [hunk 1]
```diff
@@ -24,8 +24,10 @@
         <opentracing-jdbc.version>0.2.4</opentracing-jdbc.version>
         <opentracing-kafka.version>0.1.15</opentracing-kafka.version>
         <opentracing-mongo.version>0.1.5</opentracing-mongo.version>
-        <opentelemetry.version>1.19.0</opentelemetry.version>
-        <opentelemetry-alpha.version>1.19.0-alpha</opentelemetry-alpha.version>
+        <opentelemetry.version>1.21.0</opentelemetry.version>
+        <opentelemetry-alpha.version>1.21.0-alpha</opentelemetry-alpha.version>
+        <opentelemetry-aws.contrib.version>1.20.1-alpha</opentelemetry-aws.contrib.version>
+        <opentelemetry-aws-xray.contrib.version>1.20.1</opentelemetry-aws-xray.contrib.version>
```
[the remark this answers is not in the document]
Comment (brunobat) [reply bom/application/pom.xml:30]: They need to be moved out of core. Created this: https://github.com/quarkusio/quarkus/issues/29879

**Comment by gsmet** [comment]
We usually don't backport minor version bumps. Is there a specific reason to do so in this case?

**Comment by brunobat** [comment]
@gsmet we can live with it only in main. Thanks

**Comment by michalvavrik** [comment]
@brunobat @radcortez As long as you are fine with changes, please merge PR so that I can rebase mine. Thanks

**Pull request merged**